src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        User userWithEmptyPassword = new User();
+        userWithEmptyPassword.setLogin("validLogin");
+        userWithEmptyPassword.setPassword("");
+        userWithEmptyPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_minLengthLogin_ok() {
+        User userWithMinLengthLogin = new User();
+        String minLengthLogin = "MinLog";
+        userWithMinLengthLogin.setLogin(minLengthLogin);
+        userWithMinLengthLogin.setPassword("validPassword");
+        userWithMinLengthLogin.setAge(20);
+
+        registrationService.register(userWithMinLengthLogin);
+        assertEquals(userWithMinLengthLogin, storageDao.get(minLengthLogin));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User userWithEmptyLogin = new User();
+        userWithEmptyLogin.setLogin("");
+        userWithEmptyLogin.setPassword("validPassword");
+        userWithEmptyLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyLogin);
+        });
+    }
+
+    @Test
+    void register_ageIsZero_notOk() {
+        User userWithZeroAge = new User();
+        userWithZeroAge.setLogin("validLogin");
+        userWithZeroAge.setPassword("validPassword");
+        userWithZeroAge.setAge(0);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithZeroAge);
+        });
+    }
+
+    @Test
+    void register_fewYears_notOk() {
+        User userWithSeventeenYears = new User();
+        userWithSeventeenYears.setLogin("validLogin");
+        userWithSeventeenYears.setPassword("validPassword");
+        userWithSeventeenYears.setAge(17);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithSeventeenYears);
+        });
+    }
+
+    @Test
+    void register_minAge_ok() {
+        User userWithMinAge = new User();
+        userWithMinAge.setLogin("LoginV");
+        userWithMinAge.setPassword("PasswordV");
+        userWithMinAge.setAge(18);
+        registrationService.register(userWithMinAge);
+        assertEquals(userWithMinAge, storageDao.get("LoginV"));
```

Comment:
Same as above, you should also verify that the returned User object is the same as the one you passed to the method.